towerofhanoi/Program.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace towerofhanoi
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Welcome to Tower of Hanoi");
+            Game towerofHanoiGame = new Game();
+            towerofHanoiGame.Play();
+        }
+    }
+    class Blocks
+    {
+        public int weightedBlock { get; set; }
+        public Blocks (int weightedBlock)
+        {
+            this.weightedBlock = weightedBlock;
+        }
+    }
+ class Towers {
+        public Stack<Blocks> towerBlocks { get; set; }
+        public Towers () {
+            this.towerBlocks = new Stack<Blocks> ();
+        }
+    }
+    class Game
+    {
+        //declaring variables and definitions
+       Dictionary<String, Towers> MakeTowers = new Dictionary<String, Towers>();
+        bool win = false;
+        string userChoice1 = "";
+        bool validMoveFrom = false;
+        String moveFrom = "";
+        String userChoice2 = "";
+        bool validMoveTo = false;
+        String movedTo = "";
+
+        //Game Constructor
+
+        public Game()
+        {
+            MakeTowers["A"] = new Towers();
+            MakeTowers["B"] = new Towers();
+            MakeTowers["C"] = new Towers();
+            //makes blocks for the towers
+            Blocks block1 = new Blocks(1);
+            Blocks block2 = new Blocks(2);
+            Blocks block3 = new Blocks(3);
+            Blocks block4 = new Blocks(4);
+            //adds blocks to tower A for the initial start of the game
+            MakeTowers["A"].towerBlocks.Push(block4);
+            MakeTowers["A"].towerBlocks.Push(block3);
+            MakeTowers["A"].towerBlocks.Push(block2);
+            MakeTowers["A"].towerBlocks.Push(block1);
+
+        }
+        public void PrintBoard()
+        {
+            foreach (String name in MakeTowers.Keys)
+            {
+                System.Console.WriteLine();
+                System.Console.WriteLine(name + ": ");
+                Towers tower = MakeTowers[name];
+                Stack<Blocks> stackofBlocks = tower.towerBlocks;
+                Stack<Blocks> stackPrinted = new Stack<Blocks>();
+
+                foreach (Blocks b in stackofBlocks)
+                {
+                    stackPrinted.Push(b);
+                    foreach (Blocks z in stackPrinted)
+                    {
+                        System.Console.WriteLine(z.weightedBlock);
+                    }
+                    System.Console.WriteLine();
+                }
+            }
+        }
+        public string UserMoveFrom()
+        {
+            Console.WriteLine("What tower are you moving your block from? Please choose 'A'  'B' or 'C' ");
+            userChoice1 = Console.ReadLine().ToUpper().Trim();
+            return userChoice1;
+        }
+        //validates choice
+        public bool ValidMoveFrom(string userChoice1)
+        {
+            bool validMoveFrom;
```

Comment:
This is the root of first your problem. By declaring this variable with the same name as your class level field, you shadow (hide) access to the field validMoveFrom with the local variable. This is why you should always use `this` when accessing properties in your class so there is no confusion. If you remove this, you get ride of your never ending loop that causes your program to stop responding.